### DISCUSSION TITLE
isort known first-party modules starting with a string

### DISCUSSION THREAD

**Discussion by fbaldi6** [astral-sh/ruff#5419] [Q&A]
With isort, I could use the following configuration to specify that all modules starting with `mymodule_` should be considered as "known first-party".

```
[tool.isort]
known_first_party = ["mymodule_*"]
```

How can I achieve the same with ruff?

**Comment by tjkuson** [comment]
This option has been translated to ruff. 

```
[tool.ruff.isort]
known-first-party = ["mymodule_*"]
```

The documentation for the option exists [here](https://beta.ruff.rs/docs/settings/#isort-known-first-party). Hopefully this works!

**Reply by fbaldi6** [reply]
First of all, thank you for your response! The option has been translated as you rightly pointed out. However, somehow using the asterisk `*` to indicate "all modules starting with that string" does not seem to work.

**Reply by charliermarsh** [reply]
I think the one distinction is that we don't support globs.

**Reply by tjkuson** [reply]
Oh, sorry about that. Would this be feasible to implement?

**Reply by charliermarsh** [reply]
Yeah, we can create an issue for it. We support globbing for some settings, but not all. In the meantime, the suggested workaround would be to just enumerate the modules in the setting :joy:

**Reply by fbaldi6** [reply]
> we can create an issue for it

That would be great, thank you!

> In the meantime, the suggested workaround would be to just enumerate the modules in the setting 😂

I know, but unfortunately with a hundred or so ever-expanding modules it becomes quite cumbersome 😅

**Reply by charliermarsh** [reply]
Totally understand. Is there any way to get Ruff to pick up your modules without using `known_first_party`? E.g., have you considered adding the parent directory as a source via the [`src`](https://beta.ruff.rs/docs/settings/#src) setting?

**Reply by tjkuson** [reply]
@fbaldi6 #5473 was merged, so hopefully this should be working in the next release (0.0.278)!

**Comment by saippuakauppias** [comment]
Its related to https://github.com/astral-sh/ruff/issues/5667 ?
